compiler/rustc_trait_selection/src/error_reporting/traits/mod.rs

```diff
@@ -162,15 +172,42 @@ impl<'a, 'tcx> TypeErrCtxt<'a, 'tcx> {
 
         // Ensure `T: Sized` and `T: WF` obligations come last. This lets us display diagnostics
         // with more relevant type information and hide redundant E0282 errors.
-        errors.sort_by_key(|e| match e.obligation.predicate.kind().skip_binder() {
-            ty::PredicateKind::Clause(ty::ClauseKind::Trait(pred))
-                if self.tcx.is_lang_item(pred.def_id(), LangItem::Sized) =>
-            {
-                1
+        errors.sort_by_key(|e| {
+            let span = e.obligation.cause.span;
+            let outer_expn_data = span.ctxt().outer_expn_data();
+
+            match e.obligation.predicate.kind().skip_binder() {
+                ty::PredicateKind::Subtype(_)
+                    if matches! (
+                        outer_expn_data.kind,
+                        ExpnKind::Macro(_, name) if matches!(
+                            name.as_str().rsplit("::").next(),
+                            Some("format_args" | "format_args_nl")
```

Comment:
I think you may be able to use is_diagnostic_item for this, and mark `format_args_nl` as a diagnostic item.